src/main/java/net/dries007/tfc/common/blockentities/RottingBlockEntity.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.blockentities;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+
+import net.dries007.tfc.util.calendar.Calendars;
+
+public class RottingBlockEntity extends TFCBlockEntity
```

Comment:
As this is just `TickCounterBlockEntity` but with a different name / helper methods, I think it might just be cleaner to have all the rotting specific logic in the block, and then use the tick counter block entity.